src/node_sockaddr.h

```diff
@@ -0,0 +1,109 @@
+#ifndef SRC_NODE_SOCKADDR_H_
+#define SRC_NODE_SOCKADDR_H_
+
+#if defined(NODE_WANT_INTERNALS) && NODE_WANT_INTERNALS
+
+#include "env.h"
+#include "node.h"
+#include "uv.h"
+#include "v8.h"
+
+#include <string>
+
+namespace node {
+
+class SocketAddress {
+ public:
+  struct Hash {
+    inline size_t operator()(const sockaddr* addr) const;
+  };
+
+  struct Compare {
+    inline bool operator()(const sockaddr* laddr, const sockaddr* raddr) const;
+  };
+
+  inline static bool is_numeric_host(const char* hostname);
+
+  inline static bool is_numeric_host(const char* hostname, int family);
+
+  inline static sockaddr_storage* ToSockAddr(
+      int32_t family,
+      const char* host,
+      uint32_t port,
+      sockaddr_storage* addr);
+
+  inline static int GetPort(const sockaddr* addr);
+
+  inline static int GetPort(const sockaddr_storage* addr);
+
+  inline static std::string GetAddress(const sockaddr* addr);
+
+  inline static std::string GetAddress(const sockaddr_storage* addr);
+
+  inline static size_t GetLength(const sockaddr* addr);
+
+  inline static size_t GetLength(const sockaddr_storage* addr);
+
+  SocketAddress() {}
+
+  inline explicit SocketAddress(const sockaddr* addr);
+
+  inline explicit SocketAddress(const SocketAddress& addr);
+
+  inline SocketAddress& operator=(const sockaddr* other);
+
+  inline SocketAddress& operator=(const SocketAddress& other);
+
+  inline const sockaddr* operator*() const;
```

Comment:
Just a side note… it’s untypical to return a pointer from `operator*`; more commonly, it would be a reference or const reference + a pointer-returning `operator->`, like what `NgRcBufPointer` or `BaseObjectPtrImpl` do

Reply:
Noted... we do this in a few places... I'll go back and clean those up in a separate PR